client/reader/list-manage/index.jsx

```diff
@@ -40,86 +45,112 @@ function ReaderListEdit( props ) {
-									accessible to everyone.
-								</FormSettingExplanation>
-							</FormFieldset>
+										<FormLabel>
+											<FormRadio value="private" checked={ ! list.is_public } />
+											<span>Only I can view this list</span>
+										</FormLabel>
+										<FormSettingExplanation>
+											Don't worry, posts from private sites will only appear to those with access.
+											Adding a private site to a public list will not make posts from that site
+											accessible to everyone.
+										</FormSettingExplanation>
+									</FormFieldset>
 
-							<FormFieldset>
-								<FormLabel htmlFor="list-description">Description</FormLabel>
-								<FormTextarea
-									name="list-description"
-									id="list-description"
-									placeholder="What's your list about?"
-									value={ list.description }
-								/>
-							</FormFieldset>
-							<FormButtonsBar>
-								<FormButton primary>Save</FormButton>
-							</FormButtonsBar>
-						</Card>
+									<FormFieldset>
+										<FormLabel htmlFor="list-description">Description</FormLabel>
+										<FormTextarea
+											name="list-description"
+											id="list-description"
+											placeholder="What's your list about?"
+											value={ list.description }
+										/>
+									</FormFieldset>
+									<FormButtonsBar>
+										<FormButton primary>Save</FormButton>
+									</FormButtonsBar>
+								</Card>
 
-						<Card>
-							<FormSectionHeading>Sites in List</FormSectionHeading>
-							<pre>
-								{ ! listItems && 'Loading...' }
-								{ listItems &&
-									listItems.map( ( item ) => JSON.stringify( item, null, 2 ) + '\n\n' ) }
-							</pre>
-						</Card>
-
-						<Card>
-							<FormSectionHeading>DANGER!!</FormSectionHeading>
-							<Button scary primary>
-								DELETE LIST FOREVER
-							</Button>
-						</Card>
+								<Card>
+									<FormSectionHeading>DANGER!!</FormSectionHeading>
+									<Button scary primary>
+										DELETE LIST FOREVER
+									</Button>
+								</Card>
+							</>
+						) }
+						{ selectedSection === 'sites' &&
+							props.sites?.map( ( site ) => (
+								<Card className="list-manage__site-card">
+									{ /* Shares same index order as props.listItems */ }
+									{ ! site ? (
+										<SitePlaceholder />
+									) : (
+										<>
+											<Site site={ site } />
+											<Button scary primary>
+												Remove from list
+											</Button>
+										</>
+									) }
+								</Card>
+							) ) }
 					</>
 				) }
 			</Main>
 		</>
 	);
 }
 
+const bindStateToGetSiteFn = ( state ) => ( item ) => getSite( state, item.site_ID );
 export default connect( ( state, ownProps ) => {
 	const list = getListByOwnerAndSlug( state, ownProps.owner, ownProps.slug );
 	const listItems = list ? getListItems( state, list.ID ) : undefined;
 	return {
 		list,
 		listItems,
+		getSite: bindStateToGetSiteFn( state ),
```

Comment:
items either have a site id, a feed id, or a tag, but never more than one.

Reply:
anyway, we should push this bit of the connect down to a new set of `Site`, `Feed`, and `Tag` components. Let them deal with loading the correct info and showing it.

Reply:
👍